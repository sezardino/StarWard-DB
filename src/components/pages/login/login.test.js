import React from 'react';
import {shallow} from 'enzyme';

import LoginPage from './login';

describe('Component LoginPage', () => {
	const onLogin = jest.fn();
	const component = shallow(<LoginPage onLogin={onLogin} />);
	it('Component rendered correctly', () => {
		expect(component.debug()).toMatchSnapshot();
	});

	describe('Components handlers', () => {
		it('Click on login button call onLogin', () => {
			const loginButton = component.find('button');
			expect(onLogin).toBeCalledTimes(0);
			loginButton.simulate('click');
			expect(onLogin).toBeCalledTimes(1);
		});
	});
});
