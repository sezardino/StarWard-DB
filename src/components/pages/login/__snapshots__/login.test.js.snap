// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LoginPage Component rendered correctly 1`] = `
"<div className=\\"jumbotron\\">
  <p>
    Login to see secret page!
  </p>
  <button className=\\"btn btn-primary\\" onClick={[Function: mockConstructor]}>
    Login
  </button>
</div>"
`;
